### PR TITLE
Add Nial Lexer

### PR DESCRIPTION
**Pull request by razetime**
For use in the main Nial language website: https://nial-array-language.org/

### CODE REVIEW

File: lib/rouge/lexers/nial.rb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                                       "endfor", "while", "do", "endwhile",
+                                       "repeat", "until", "endrepeat"]
+            end
+
+            def self.operators
+                @operators||= Set.new [".",  "!", "#", "+", "*", "-", "<<",
+                                 "/", "<", ">>", "<=", ">", "=", ">=", "@", "|", "~="]
+            end
+
+            def self.punctuation
```
Comment (tancnle) [inline lib/rouge/lexers/nial.rb:25]: **suggestion:* Could we rename this to a plural name i.e, `punctuations`?

File: lib/rouge/lexers/nial.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+# frozen_string_literal: true
+
+module Rouge
+    module Lexers
+        class Nial < RegexLexer
+            title 'Nial'
+            desc 'The Nial programming language (nial-array-language.org)'
+            tag 'nial'
+            filenames '*.ndf', '*.nlg'
+            def self.keywords
```
Comment (tancnle) [inline lib/rouge/lexers/nial.rb:11]: **nitpick:** Could we add an empty line between L10 and L11?

**Review by tancnle — COMMENTED** [review]
Thank you for your contribution @razetime ❤️ The PR looks good to me overall. 

It seems that the CI hasn't run on this PR yet. Could you kindly rebase this off the master branch?

We use a 2-space indentation for Ruby in this code base. Could you fix this?

File: lib/rouge/lexers/nial.rb (hunk 1 above)
Comment (razetime) [reply lib/rouge/lexers/nial.rb]: renamed.

File: lib/rouge/lexers/nial.rb (hunk 2 above)
Comment (razetime) [reply lib/rouge/lexers/nial.rb:12]: added.

**Comment by razetime** [comment]
i've done the rebase. I hope this is correct.

**Comment by tancnle** [comment]
Thanks, @razetime. It seems that the rebase has gone awry with all the extra commits that are already in the upstream master branch. Generally, we would create a feature branch on your fork instead of using the `master` branch. Few options:

1. Create a feature branch on your fork and cherry pick relevant commits. Close this PR and create a new one tracking the new feature branch. (preferred)
2. Realign `razetime/master` with `rouge-ruby/master` (WARNING: this might blow away other changes on your local fork).
    1. Reset to earlier commit to exclude non-relevant commits
       ```sh
       git reset --hard bb5c7d2
       ```
    2. Fetch upstream and rebase
        ```sh
        git fetch upstream
        git rebase upstream/master
        ```
    3. Add the last missing commit
       ```sh
       git cherry-pick e09905c
       ```
    4. Force push to origin
        ```sh
        git push -f origin master
        ```

**Comment by razetime** [comment]
Continued in #1844.

**Pull request closed without merging**